### PR TITLE
ISPN-8709 REST server not properly caching internal instances

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-8709

### CODE REVIEW

**Comment by gustavonalle** [comment]
CI is fine

**Comment by gustavonalle** [comment]
Can someone review this please? @slaskawi @wburns @galderz ?

File: server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java [hunk 1]
```diff
@@ -43,62 +43,50 @@
...
       this.instance = instance;
       this.isCacheIgnored = isCacheIgnored;
+      this.icr = instance.getGlobalComponentRegistry().getComponent(InternalCacheRegistry.class);
       this.allowInternalCacheAccess = instance.getCacheManagerConfiguration().security().authorization().enabled();
    }
 
    @SuppressWarnings("unchecked")
    public AdvancedCache<String, V> getCache(String name, MediaType requestedMediaType) {
+      if (isCacheIgnored.test(name)) {
+         throw new ServiceUnavailableException("Cache with name '" + name + "' is temporarily unavailable.");
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java]: i18n

File: server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java [hunk 2]
```diff
@@ -43,62 +43,50 @@
...
+      }
       if (requestedMediaType == null) {
          throw new NullPointerException("requestedMediaType cannot be null");
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java]: i18n

**Comment by gustavonalle** [comment]
Added i18n

**Comment by gustavonalle** [comment]
run performance tests please
cs-rest

**Comment by gustavonalle** [comment]
From the perf results, the regression is fixed

**Comment by anistor** [comment]
Integrated in master. Thanks @gustavonalle !

**Pull request closed without merging**